src/Dictionaries/DictionaryHelpers.h

```diff
@@ -600,6 +600,7 @@ void mergeBlockWithPipe(
     while (executor.pull(block))
     {
         convertToFullIfSparse(block);
+        block.checkNumberOfRows();
```

Comment:
Yes, if some column is nullptr, it will crash here https://github.com/ClickHouse/ClickHouse/pull/72734/files#diff-6df141490c9121400324e65a1e58b5026e7ac74df20fe78f194866bf30b1e1eeR481